crates/red_knot/src/main.rs

```diff
@@ -132,28 +129,43 @@ fn run() -> anyhow::Result<ExitStatus> {
     countme::enable(verbosity.is_trace());
     let _guard = setup_tracing(verbosity)?;
 
-    let cwd = if let Some(cwd) = current_directory {
-        let canonicalized = cwd.as_utf8_path().canonicalize_utf8().unwrap();
-        SystemPathBuf::from_utf8_path_buf(canonicalized)
-    } else {
-        let cwd = std::env::current_dir().unwrap();
-        SystemPathBuf::from_path_buf(cwd).unwrap()
+    // The base path to which all CLI arguments are relative to.
+    let cli_base_path = {
+        let cwd = std::env::current_dir().context("Failed to get the current working directory")?;
+        SystemPathBuf::from_path_buf(cwd).map_err(|path| anyhow!("The current working directory '{}' contains non-unicode characters. Red Knot only supports unicode paths.", path.display()))?
```

Comment:
Nit: `s/unicode/UTF-8/`.

We have no idea if it contains unicode or not, because unicode is just a set of numeric code points, it's not an encoding. It may well be unicode encoded as UTF-32, for all we know. It's just not valid UTF8-encoded text.

Reply:
I'm not so sure if we're limited to UTF8 here. [`Path::to_string_lossly`](https://doc.rust-lang.org/stable/std/ffi/struct.OsString.html#method.to_string_lossy) specifically mentioned unicode and not UTF8

Reply:
It only handles UTF-8. It's not possible to transparently and automatically handle all the various Unicode encodings; they are ambiguous with each other. You have to either assume or specify a particular encoding. 

It's too bad that the rust docs are also careless with this wording. It's meaningless to say that a byte string "is Unicode" or "is not Unicode", unless by "Unicode" you mean (or assume) some specific encoding.

Reply:
Yeah, still not sure about this. It's true that you can only store one encoding, but you can decode between encodings. E.g. Rust converts the Windows UTF16 paths to UTF8 

Also, from the camino docs https://docs.rs/camino/latest/camino/

Reply:
I see. You (and probably the rust docs too) don't want to say UTF-8 (even though the actual decode that's failing is definitely a decode from UTF-8, or the windows "wide" variant of it) because the original path on disk on windows is UTF-16 (decoded by rust and re-encoded as UTF-8 in the OsStr). So we use "Unicode" as imprecise shorthand for "the typical Unicode encoding used on your OS, ie UTF-16 on Windows and UTF-8 everywhere else." I guess maybe that's reasonable, for lack of a more precise term.